plugins/autosuspend.koplugin/main.lua

```diff
@@ -421,17 +446,22 @@ end
 -- UI signals us that standby is allowed at this very moment because nothing else goes on in the background.
 function AutoSuspend:onAllowStandby()
     logger.dbg("AutoSuspend: onAllowStandby")
-    -- In case the OS frontend itself doesn't manage power state, we can do it on our own here.
-    -- One should also configure wake-up pins and perhaps wake alarm,
-    -- if we want to enter deeper sleep states later on from within standby.
+    -- This piggy-backs minimally on the UI framework implemented for the PocketBook autostandby plugin,
+    -- see its own AllowStandby handler for more details.
+
+    -- Don't enter standby if we haven't set a proper timeout yet.
+    if not self:_enabledStandby() then
+        logger.dbg("AutoSuspend: No timeout set, no standby")
+        return
+    end
 
-    -- Don't enter standby if wifi is on, as this my break reconnecting (at least on Kobo-Sage)
+    -- Don't enter standby if wifi is on, as this will break in fun and interesting ways (from Wi-Fi issues to kernel deadlocks).
```

Comment:
Those three guards (if ...) could go into one if elseif elseif block.

Reply:
They're not mutually exclusive though ;).